components/BridgeTokenDestination.tsx

```diff
@@ -0,0 +1,185 @@
+import { useCENNZWallet } from "@/providers/CENNZWalletProvider";
+import { useMetaMaskWallet } from "@/providers/MetaMaskWalletProvider";
+
+interface BridgeTokenDestinationProps {}
+
+const BridgeTokenDestination: VFC<
+	IntrinsicElements["div"] & BridgeTokenDestinationProps
+> = (props) => {
+	const {
+		transferSelect,
+		transferInput,
+		ethereumTokens,
+		transferAsset,
+		bridgeAction,
+		transferCENNZAddress,
+		setTransferCENNZAddress,
+		transferMetaMaskAddress,
+		setTransferMetaMaskAddress,
+		metaMaskBalance,
+	} = useBridge();
+
+	const [cennzBalance] = useCENNZBalances(
+		transferAsset as BridgedEthereumToken
+	);
+
+	const transferBalance =
+		bridgeAction === "Withdraw" ? cennzBalance : metaMaskBalance;
+
+	const onTransferMaxRequest = useMemo(() => {
+		const setErc20Value = transferInput.setValue;
+		return () => setErc20Value(transferBalance.toInput());
+	}, [transferBalance, transferInput.setValue]);
+
+	const onTransferCENNZAddressChange = useCallback(
+		(event) => {
+			setTransferCENNZAddress(event.target.value);
+		},
+		[setTransferCENNZAddress]
+	);
+
+	const onTransferMetaMaskAddressChange = useCallback(
+		(event) => {
+			setTransferMetaMaskAddress(event.target.value);
+		},
+		[setTransferMetaMaskAddress]
+	);
+
+	const { inputRef: transferInputRef } = useBalanceValidation(
+		Balance.fromInput(transferInput.value, transferAsset),
+		transferBalance
+	);
+
+	const { inputRef: cennzAddressInputRef } = useAddressValidation(
+		transferCENNZAddress,
+		"CENNZnet"
+	);
+
+	const { inputRef: metaMaskAddressInputRef } = useAddressValidation(
+		transferMetaMaskAddress,
+		"Ethereum"
+	);
+
+	// Sync CENNZ selected account with `transferCENNZAddressInput`
+	const { selectedAccount: cennzAccount } = useCENNZWallet();
+	useEffect(() => {
+		if (!cennzAccount?.address) return;
+		setTransferCENNZAddress(cennzAccount.address);
+	}, [cennzAccount?.address, setTransferCENNZAddress]);
+
+	// Synce MetaMask selected account with `transferMetaMaskAddressInput`
+	const { selectedAccount: metaMaskAccount } = useMetaMaskWallet();
+	useEffect(() => {
+		if (!metaMaskAccount?.address) return;
+		setTransferMetaMaskAddress(metaMaskAccount.address);
+	}, [metaMaskAccount?.address, setTransferMetaMaskAddress]);
+
+	return (
+		<div {...props} css={styles.root}>
+			<div css={styles.formField}>
+				<label htmlFor="transferInput">TRANSFER TOKEN</label>
+				<TokenInput
+					onMaxValueRequest={onTransferMaxRequest}
+					selectedTokenId={transferSelect.tokenId}
+					onTokenChange={transferSelect.onTokenChange}
+					value={transferInput.value}
+					onValueChange={transferInput.onValueChange}
+					tokens={ethereumTokens}
+					id="transferInput"
+					ref={transferInputRef}
+					required
+					scale={4}
+					min={0.0001}
+				/>
+				{transferBalance !== null && (
+					<div css={styles.tokenBalance}>
+						Balance: <span>{transferBalance?.toBalance() ?? "0.0000"}</span>
+					</div>
```

Comment:
cool ?? syntax

Reply:
👍  it's https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing_operator